### DISCUSSION TITLE
[next-auth]: useSession must be wrapped in a <SessionProvider /> pages folder move to inside src folder next13 app dir

### DISCUSSION THREAD

**Discussion by liamtran96** [nextauthjs/next-auth#7747] [Help]
### Provider type

Credentials

### Environment

System:
    OS: macOS 13.3.1
    CPU: (4) x64 Intel(R) Core(TM) i5-7267U CPU @ 3.10GHz
    Memory: 261.21 MB / 8.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 16.15.0 - /usr/local/bin/node
    Yarn: 1.22.19 - ~/.yarn/bin/yarn
    npm: 9.6.2 - /usr/local/bin/npm
  Browsers:
    Chrome: 114.0.5735.90
    Firefox: 112.0.2
    Safari: 16.4

### Reproduction URL

local development

### Describe the issue

I am using next 13.4.0 app dir  with nextauth 4.22.1 for my authentication but I got the error [next-auth]: useSession must be wrapped in a <SessionProvider /> pages folder move to inside src folder on my console 
<img width="1440" alt="Screenshot 2023-06-05 at 7 57 14 AM" src="https://github.com/nextauthjs/next-auth/assets/111833625/b6d7df23-161e-4873-b8aa-4c5052a721e8">
There was some suggestions here #3061  it works permanently, the error still there when i start my project when i use: yarn dev

### How to reproduce

So this is my config with next auth Credentials
<img width="1440" alt="Screenshot 2023-06-05 at 8 01 09 AM" src="https://github.com/nextauthjs/next-auth/assets/111833625/2b8e3573-cfc5-4ab2-8a6c-7aa5207e186c">

Provider
<img width="1440" alt="Screenshot 2023-06-05 at 8 01 35 AM" src="https://github.com/nextauthjs/next-auth/assets/111833625/e42a63a4-ded7-4a7f-8b49-06c14611fdff">

authOptions
<img width="1440" alt="Screenshot 2023-06-05 at 8 03 38 AM" src="https://github.com/nextauthjs/next-auth/assets/111833625/076d6a3e-bea8-488d-9474-d657119d05c9">

<img width="1440" alt="Screenshot 2023-06-05 at 8 04 04 AM" src="https://github.com/nextauthjs/next-auth/assets/111833625/0b1b3825-7bbb-421e-aa8d-3dfac5c09978">

I think there are something wrong with the layout.tsx file
<img width="1440" alt="Screenshot 2023-06-05 at 8 04 25 AM" src="https://github.com/nextauthjs/next-auth/assets/111833625/a3c3d5b3-849a-4acf-9a82-972c0616034d">

I use the useSession in my login page 
<img width="1440" alt="Screenshot 2023-06-05 at 8 05 51 AM" src="https://github.com/nextauthjs/next-auth/assets/111833625/19678b45-7796-453a-ac37-365686ae1d0a">

### Expected behavior

How can I fix the error, many thanks!!

**Comment by imyuanli** [comment]
I also encountered the same problem, my nextjs version is 13.4.4

**Reply by liamtran96** [reply]
https://github.com/nextauthjs/next-auth/issues/4696

You can solve the problem by change import const to export default function. Have a nice day.

**Reply by sogeking7** [reply]
Didn't work me

**Comment by messiisgreat** [comment]
I also met that kind of error 

Error: [next-auth]: `useSession` must be wrapped in a <SessionProvider />

![image](https://github.com/nextauthjs/next-auth/assets/134148939/6e2e42ce-6a77-4700-93c9-34870e3537df)

**Reply by mateodevcode** [reply]
Como lo resolviste?

**Reply by messiisgreat** [reply]
I tryed sessionProvider

```
"use client";
import React from "react";
import { SessionProvider } from "next-auth/react";

const AuthProvider = ({ children }) => {
  return <SessionProvider>{children}</SessionProvider>;
};

export default AuthProvider;
```

**Comment by rrdukeakd** [comment]
Create a new file "SessionWrapper"
Paste the code below 
```
"use client";

import { SessionProvider } from "next-auth/react";

const SessionWrapper = ({ children }) => {
  return <SessionProvider>{children}</SessionProvider>;
};

export default SessionWrapper;

```

import it in layout.js file and wrap the  body with it.
